miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -271,6 +270,11 @@ private Sample save(Sample sample) throws IOException {
         created.setAlias(created.getName());
         needsUpdate = true;
       }
+      if (autoGenerateIdBarcodes && isStringEmptyOrNull(created.getIdentificationBarcode())) {
```

Comment:
Can probably happen before initial save, so it doesn't cause a resave to be required